src/gui/widgets/options_button.hpp

```diff
@@ -110,15 +122,27 @@ class options_button : public styled_widget
 	 */
 	state_t state_;
 
+//protected:
 	boost::container::stable_vector<menu_item> values_;
 
-protected:
+	/* toggle_states_ is used only to transfer checkbox values to and from the widget, and may
+	 * not be up to date at any other time.  Use values_ for all other operations */
+    boost::dynamic_bitset<> toggle_states_;
```

Comment:
This sounds suspicious. Why do you need a transient class member for that purpose?

Reply:
I'm not sure what a "transient class member" is, but...

toggle_states_ was there originally.  I found it to be a bit confusing since it was redundant with the data in values_.  But then when I added random (non-index based) access to the vector via add_item and the ability to remove rows it became IMO more trouble than it was worth.  I'd like to keep the bitset concept for transferring checkbox values, but lose the member variable, but I wanted to get some advice first.

Reply:
Hmm, so you're saying you weren't the one who put this variable there. I think it would make more sense for it not to be there, but perhaps it's an optimization…